sonar-python-plugin/src/test/java/org/sonar/plugins/python/PythonHighlighterTest.java

```diff
@@ -0,0 +1,119 @@
+import org.sonar.api.batch.fs.internal.FileMetadata;
+import org.sonar.api.batch.sensor.highlighting.TypeOfText;
+import org.sonar.api.batch.sensor.internal.SensorContextTester;
+import org.sonar.python.PythonAstScanner;
+
+import static org.fest.assertions.Assertions.assertThat;
+
+public class PythonHighlighterTest {
+  
+  private SensorContextTester context;
+  
+  private File file; 
+  
+  @Before
+  @SuppressWarnings("unchecked")
+  public void scanFile() {
+    file = new File("src/test/resources/org/sonar/plugins/python/highlighting.py");
+    DefaultInputFile inputFile = new DefaultInputFile("moduleKey", file.getName())
+      .initMetadata(new FileMetadata().readMetadata(file, Charsets.UTF_8));
+    
+    context = SensorContextTester.create(new File("src/test/resources/org/sonar/plugins/python").getAbsoluteFile());
+    context.fileSystem().add(inputFile);
+    
+    PythonHighlighter pythonHighlighter = new PythonHighlighter(context);
+    PythonAstScanner.scanSingleFile(file, pythonHighlighter);
+  }
+
+  @Test
+  public void keyword_def() throws Exception {
+    List<TypeOfText> typeOfTexts = getTypesOfText(1, 1, 1);
+
+    assertThat(typeOfTexts.get(0)).isEqualTo(TypeOfText.KEYWORD);
+  }
+
+  @Test
+  public void keyword_four() throws Exception {
+    List<TypeOfText> typeOfTexts;
+    
+    // if
+    typeOfTexts = getTypesOfText(12, 1, 1);
+    assertThat(typeOfTexts.get(0)).isEqualTo(TypeOfText.KEYWORD);
+    
+    // or
+    typeOfTexts = getTypesOfText(12, 13, 1);
+    assertThat(typeOfTexts.get(0)).isEqualTo(TypeOfText.KEYWORD);
+    
+    // or
+    typeOfTexts = getTypesOfText(12, 25, 1);
+    assertThat(typeOfTexts.get(0)).isEqualTo(TypeOfText.KEYWORD);
+    
+    // continue
+    typeOfTexts = getTypesOfText(12, 38, 1);
+    assertThat(typeOfTexts.get(0)).isEqualTo(TypeOfText.KEYWORD);
+  }
+  
+  @Test
+  public void keyword_pass() throws Exception {
+    List<TypeOfText> typeOfTexts = getTypesOfText(2, 4, 1);
+    
+    assertThat(typeOfTexts.get(0)).isEqualTo(TypeOfText.KEYWORD);
+  }
+  
+  @Test
+  public void string_literal() throws Exception {
+    List<TypeOfText> typeOfTexts = getTypesOfText(4, 5, 1);
+    
+    assertThat(typeOfTexts.get(0)).isEqualTo(TypeOfText.STRING);
+  }
+  
+  @Test
+  public void comment() throws Exception {
+    List<TypeOfText> typeOfTexts = getTypesOfText(6, 6, 1);
+    
+    assertThat(typeOfTexts.get(0)).isEqualTo(TypeOfText.COMMENT);
+  }
+  
+  @Test
+  public void comment_misplaced() throws Exception {
+    List<TypeOfText> typeOfTexts = getTypesOfText(9, 18, 1);
+    
+    assertThat(typeOfTexts.get(0)).isEqualTo(TypeOfText.COMMENT);
+  }
+  
+  private List<TypeOfText>  getTypesOfText(int line, int column, int expectedSize) {
+    List<TypeOfText> typeOfTexts = context.highlightingTypeAt("moduleKey:" + file.getName(), line, column);
+
+    assertThat(typeOfTexts).as("types of text").hasSize(1);
+    
+    return typeOfTexts;
```

Comment:
@ivandalbosco You better return the first element of this list, then it will be easier to use this method